Kernel/Interrupts/SharedIRQHandler.h

```diff
@@ -27,7 +28,10 @@ class SharedIRQHandler final : public GenericInterruptHandler {
 
     void enumerate_handlers(Function<void(GenericInterruptHandler&)>&);
 
-    virtual size_t sharing_devices_count() const override { return m_handlers.size_slow(); }
+    virtual size_t sharing_devices_count() const override
+    {
+        return m_handlers.with([](auto& list) { return list.size_slow(); });
+    }
```

Comment:
Looks to me like this API is only ever used to check if the list is empty.
If that's true, we can change it to be an "is empty?" API and implement it with an O(1) emptiness check instead of an O(n) length check on the linked list :)

Reply:
We do use it for debug statements and in `SysFSInterrupts::try_generate(KBufferBuilder& builder)`. The only other place where the emptiness check is done is in `unregister_generic_interrupt_handler`, which is not in the hot path anyway.